app/src/main/java/com/example/geoquiz/QuizViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.geoquiz
+
+import androidx.lifecycle.ViewModel
+
+class QuizViewModel : ViewModel() {
+
+    var currentIndex: Int = 0
+    var isCheater = false
+    lateinit var correctAnswers: ArrayList<Int>
```

Comment:
As mentioned before, this list makes no sense to me. It is higher complexity than using a simple counter with no benefit.
You can use a simple integer counter to count the correct answers.

That's at least how I understand the current implementation which I think is wrong because every time checkAnswer is called, the list is emptied